### ISSUE TITLE
U2U1106 suggests less efficient code in case of const string concatenation

### ISSUE THREAD

**Issue by lechu445** [u2uconsult/codeanalyzers#31]
for example:
```c#
var sb = new StringBuilder();
sb.Append("A" + "B"); //suggests to split into two Append calls
sb.Append("A" + 1); //suggests to split into two Append calls
```
In fact the string concatenation is done at compilation time (as the values are consts), so the compiler will produce `sb.Append("AB")` which is more efficient than suggested `sb.Append("A").Append("B")`.

So, I propose to fix the analyzer to check if all concatenated with `+` sign values are consts and do not suggest refactoring in this case.

It would also be nice to have a backward analyzer that could detect such situations:
```c#
var sb = new StringBuilder();
sb.Append("A").Append("B").Append(1);
``` 
to change into:
```c#
var sb = new StringBuilder();
sb.Append("AB1");
```
It reduces Append method calls.

**Comment by KrisVandermotten** [comment]
> ```cs
> var sb = new StringBuilder();
> sb.Append("A" + "B"); //suggests to split into two Append calls
> ```

The above call should not produce a warning. What version of the analyzers are you using? Make sure you upgrade to the latest version if necessary.

> ```cs
> var sb = new StringBuilder();
> sb.Append("A" + 1); //suggests to split into two Append calls
> ```

This will produce a warning, as the expression `"A" + 1` is not a compile-time constant, and [will be calculated at runtime](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGAOgBUYEMBuAWAChiAmcgYXIDeHciMoBmSinIBZABQBKQcNEqAbtijlcwcgF5yAOxgB3cgGUMUAJYGA5gCEArlYA2AExhQFbdipXaaAEEAB2CYAzdZACJAqPIAanIqeR8VAF8ONKA). That calculation is more efficiently done using two separate `Append` calls.

> It would also be nice to have a backward analyzer that could detect such situations:

Thanks for the suggestion, we will consider it.

**Comment by lechu445** [comment]
@KrisVandermotten you are right 100%
The issue was my wrong assumption, and I double-checked it and found out that in my case one of concatenated strings were not const.
Thank you and apologies for unnecessary Issue. I am closing it.
